packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -684,6 +728,24 @@ export abstract class BaseService extends Resource
     };
   }
 
+  /**
+   *   Add an alarm based on ECS metric
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
+    // Throw an error if service connect is not configured
+    if (!this._serviceConnectConfig) {
+      throw new Error('Service connect must be enabled to set service connect metric alarms.');
+    }
+    const ecsMetric = this.metric(metric, ecsMetricAlarmProps?.metricProps);
+    const metricAlarm = new cloudwatch.Alarm(this, `${metric}Alarm`, ecsMetricAlarmProps?.useAsDeploymentAlarm && ecsMetricAlarmProps?.alarmProps ? ecsMetricAlarmProps.alarmProps : {
+      metric: ecsMetric as IMetric,
```

Comment:
You shouldn't need this type cast; a `Metric` L2 should automatically also be a `IMetric`.

Reply:
Ok gotcha, By using the built-in Names.uniqueId(construct: Construct) CDK function to pass the IDs of alarms to the CFN Service, which allows end users to access the CFN-compatible name of any construct. This will allow us to get the CFN logical IDs of passed-in alarms to the ECS Service’s deployment configuration via a lazy list producer after the construct has been initialized. 

which as shown below

`const alarmName = Names.uniqueId(this);
    const metricAlarm = new cloudwatch.Alarm(this, alarmName,
      ecsMetricAlarmProps?.useAsDeploymentAlarm && ecsMetricAlarmProps?.alarmProps ? ecsMetricAlarmProps.alarmProps : {
        metric: ecsMetric,
        threshold: ecsMetricAlarmProps?.alarmProps?.threshold || 85,
        evaluationPeriods: ecsMetricAlarmProps?.alarmProps?.evaluationPeriods || 3,
        alarmName,
      });
    if (!this.deploymentAlarms) {
      this.deploymentAlarms = {
        enable: true,
        rollback: true,
        alarmNames: [],
      };
    }
    this.deploymentAlarms.alarmNames.push(metricAlarm.node.id);`